### PR TITLE
Create test and win32 emu layer for socket()

### PR DESCRIPTION
**Pull request by maxz-lab**
Create test cases and win32 emu layer for socket() and close()

The socketclose() on win32 takes kernel handler instead file descriptor, which requires a wrapper to match posix interface.

b/302701164

Change-Id: Ib208d99ab89b4d77763362c48465d22f8893b5f9

### CODE REVIEW

File: starboard/shared/win32/posix_emu/socket.cc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+int gen_fd() {
+  return rand() % (0x7FFFFFFF) + 1;
+}
+
+struct CriticalSection {
+  CriticalSection() { InitializeCriticalSection(&critical_section_); }
+  CRITICAL_SECTION critical_section_;
+};
+
+static std::map<int, SOCKET>* m_addr = nullptr;
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/socket.cc:33]: s_addr
This is a static obect.

File: starboard/shared/win32/posix_emu/socket.cc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+static CriticalSection s_critical_section;
+
+int handle_db_put(SOCKET socket_handle) {
+  EnterCriticalSection(&s_critical_section.critical_section_);
+  if (m_addr == nullptr) {
+    m_addr = new std::map<int, SOCKET>();
+  }
+
+  int fd = gen_fd();
+  while (m_addr->find(fd) != m_addr->end()) {
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/socket.cc:43]: Why don't we just keep a counter. This random number generation and checking to find it is used can get expensive.

File: starboard/CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,11 @@ since the version previous to it.
 
 ## Version 16
 
+### Create POSIX socket API emu layer for Win32
```
Comment (y4vor) [inline starboard/CHANGELOG.md:12]: Please re-write as:

### Added standard POSIX socket/close APIs.

The standard API `socket` can be used  from `<sys/socket.h>` and
`close` can be called  to close the socket by including `<unistd.h>`.

**Review by y4vor — APPROVED** [review]

**Comment by maxz-lab** [comment]
kokoro build tvOS failed due to codesign error.

+ echo 'FAILED: codesign could not be completed. Please follow all steps in internal/starboard/darwin/tvos/shared/PRE_TEST_SETUP.md. Exiting...'
FAILED: codesign could not be completed. Please follow all steps in internal/starboard/darwin/tvos/shared/PRE_TEST_SETUP.md. Exiting...
+ exit 1
ninja: build stopped: subcommand failed.

**Pull request merged**